include/rocksdb/table.h

```diff
@@ -138,6 +141,9 @@ struct BlockBasedTableOptions {
   // Same as block_restart_interval but used for the index block.
   int index_block_restart_interval = 1;
 
+  // number of index keys per partition of indexes in a multi-level index
+  uint64_t index_per_partition = 1;
```

Comment:
Good thinking. The implementation for using number of blocks per partition was straightforward and I figured it is good enough to enable benchmarking phase and when we got good result we can revise the implementation for better configuration or more optimized implementation. Would that work?